integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,155 @@
+ */
+public final class AlluxioFuseFileSystemOpts {
+  private final String mAlluxioPath;
+  private final List<String> mLibfuseOptions;
+  private final boolean mIsDebug;
+  private final String mMountPoint;
+
+  /**
+   * Constructs an AlluxioFuseFileSystemOpts with only Alluxio cluster configuration.
+   *
+   * @param conf     Alluxio cluster configuration
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(AlluxioConfiguration conf) {
+    List<String> fuseOptions = conf.isSet(PropertyKey.FUSE_MOUNT_OPTIONS)
+        ? conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS)
+        : ImmutableList.of();
+    return new AlluxioFuseFileSystemOpts(
+        conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH),
+        conf.getString(PropertyKey.FUSE_MOUNT_POINT),
+        fuseOptions,
+        conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED)
+    );
+  }
+
+  /**
+   * Constructs an AlluxioFuseFileSystemOpts with Alluxio cluster configuration and command line.
+   * Command line input has higher precedence if a property is set both in config and command.
+   *
+   * @param conf     Alluxio cluster configuration
+   * @param fuseCliOpts     Alluxio fuse command line input
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(
+      AlluxioConfiguration conf, AlluxioFuseCliOpts fuseCliOpts) {
+    Optional<String> alluxioPathFromCli = fuseCliOpts.getMountAlluxioPath();
+    String alluxioPath = alluxioPathFromCli.orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH));
+    Optional<String> mountPointFromCli = fuseCliOpts.getMountPoint();
+    String mountPoint = mountPointFromCli.orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_POINT));
+    Optional<List<String>> libfuseOptionsFromCli = fuseCliOpts.getFuseOptions();
+    List<String> libfuseOptions;
+    if (libfuseOptionsFromCli.isPresent()) {
+      libfuseOptions = libfuseOptionsFromCli.get();
+    } else {
+      if (conf.isSet(PropertyKey.FUSE_MOUNT_OPTIONS)) {
+        libfuseOptions = conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS);
+      } else {
+        libfuseOptions = ImmutableList.of();
+      }
+    }
+    libfuseOptions = optimizeAndFormatFuseOptions(libfuseOptions);
+    boolean isDebug = conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED);
+
+    return new AlluxioFuseFileSystemOpts(alluxioPath, mountPoint, libfuseOptions, isDebug);
+  }
+
+  /**
+   * Constructs an AlluxioFuseFileSystemOpts solely for testing purpose.
+   *
+   * @param  alluxioPath
+   * @param  mountPoint
+   * @param  fuseOptions
+   * @param  isDebug
+   * @return AlluxioFuseFileSystemOpts
+   */
+  @VisibleForTesting
+  public static AlluxioFuseFileSystemOpts create(
+      String alluxioPath, String mountPoint, List<String> fuseOptions, boolean isDebug) {
+    return new AlluxioFuseFileSystemOpts(alluxioPath, mountPoint, fuseOptions, isDebug);
+  }
+
+  private AlluxioFuseFileSystemOpts(
```

Comment:
Add arguments not null check?

Reply:
I understand these parameters should be non-null, but I'm a little reluctant to do that since checking parameters not null is not a common practice in general (Just a feeling because in most places we don't do this). Is there a particular reason we want to check here?

Reply:
lol our common practice is not good!
that's why we may throw NPE in user env
The new common practice shared by @rongrong is adding null check as much as possible in constructors and static methods.

Reply:
This can help the APP caller to understand all variables should not be null and force them to obey that